FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.Core.UnitTests/Jobs/InMemoryJobStoreTests.cs

```diff
@@ -21,213 +21,175 @@
-                new DataPeriod(DateTimeOffset.MinValue, testTime),
+                TestContainerName,
+                JobStatus.Running,
+                _testResourceTypeFilters,
+                new DataPeriod(_testStartTime, _testEndTime),
                 DateTimeOffset.Now.AddMinutes(-1));
+            await blobClient.CreateJob(activeJob);
 
-            var jobStore = CreateInMemoryJobStore(blobClient);
+            activeJob.ResourceProgresses["Patient"] = "test1234";
+            activeJob.PartIds["Patient"] = 2;
+            activeJob.ProcessedResourceCounts["Patient"] = 100;
+            activeJob.SkippedResourceCounts["Patient"] = 0;
+
+            await jobStore.UpdateJobAsync(activeJob);
 
-            var schedulerSetting = await jobStore.GetSchedulerMetadata();
-            Assert.Null(schedulerSetting);
+            var updatedJob = await LoadJobFromBlob(blobClient, $"{AzureBlobJobConstants.ActiveJobFolder}/{activeJob.Id}.json");
 
-            await Assert.ThrowsAsync<ArgumentException>(() => jobStore.CompleteJobAsync(activeJob, default));
+            // Assert job is updated.
+            Assert.Equal(activeJob.Id, updatedJob.Id);
+            Assert.Equal(JobStatus.Running, updatedJob.Status);
+            Assert.Equal("test1234", activeJob.ResourceProgresses["Patient"]);
+            Assert.Equal(2, activeJob.PartIds["Patient"]);
+            Assert.Equal(100, activeJob.ProcessedResourceCounts["Patient"]);
+            Assert.Equal(0, activeJob.SkippedResourceCounts["Patient"]);
+
+            jobStore.Dispose();
         }
 
         [Fact]
-        public async Task GivenANewJob_WhenUpdateJob_JobShouldBeUpdated()
+        public async Task GivenAnActiveJob_WhenCompleteJob_ArgumentExceptionShouldBeThrown()
         {
             var blobClient = new InMemoryBlobContainerClient();
+            var jobStore = CreateInMemoryJobStore(blobClient);
             var activeJob = new Job(
-                ContainerName,
+                TestContainerName,
                 JobStatus.Running,
-                new List<string> { "Patient", "Observation" },
-                new DataPeriod(DateTimeOffset.MinValue, DateTimeOffset.MaxValue),
+                _testResourceTypeFilters,
+                new DataPeriod(_testStartTime, _testEndTime),
                 DateTimeOffset.Now.AddMinutes(-1));
+            await blobClient.CreateJob(activeJob);
 
-            var jobStore = CreateInMemoryJobStore(blobClient);
-            await jobStore.UpdateJobAsync(activeJob, default);
-
-            var savedJob = (await jobStore.GetActiveJobsAsync()).First();
-            Assert.Equal(activeJob.Id, savedJob.Id);
+            await Assert.ThrowsAsync<ArgumentException>(() => jobStore.CompleteJobAsync(activeJob));
+            jobStore.Dispose();
         }
 
         [Fact]
-        public async Task GivenAnExistingJob_WhenUpdateJob_JobShouldBeUpdated()
+        public async Task GivenASucceededJob_WhenCompleteJob_CompletedJobShouldPresent()
```

Comment:
Can we add a test for failed job as well?

Reply:
Added. Thanks